File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.py
def solution(s):
    answer = 0
    
    def expand(f, l):
        while f >= 0 and l < len(s) and s[f] == s[l]:
            f -= 1
            l += 1
        return l - f - 1
    
    for i in range(len(s)):
        # 홀수
        answer = max(answer, expand(i, i))
        
        # 짝수
        answer = max(answer, expand(i, i + 1))
    
    return answer

File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
class Solution
{
    public int solution(String s)
    {
        int answer = 0;
        
        for (int i = 0; i < s.length(); i++) {
            // 홀수
            answer = Math.max(answer, expand(s, i, i));
            
            // 짝수
            answer = Math.max(answer, expand(s, i, i + 1));
        }

        return answer;
    }
    
    public int expand(String s, int f, int l)
    {
        while (f >= 0 && l < s.length() && s.charAt(f) == s.charAt(l)) {
            f--;
            l++;
        }
        
         return l - f - 1;
    }
}